crates/sui-indexer/src/restorer/formal_snapshot.rs

```diff
@@ -0,0 +1,246 @@
+                            LiveObjectIter::new(&file_metadata, bytes.clone()).map(|obj_iter| {
+                                for object in obj_iter {
+                                    match object {
+                                        LiveObject::Normal(obj) => {
+                                            if !obj.is_package() {
+                                                let indexed_object = IndexedObject::from_object(
+                                                    next_checkpoint_after_epoch,
+                                                    obj,
+                                                    None,
+                                                );
+                                                move_objects.push(indexed_object);
+                                            }
+                                        }
+                                        LiveObject::Wrapped(_) => {}
+                                    }
+                                }
+                            });
+
+                        let live_obj_cnt = move_objects.len();
+                        let object_changes = TransactionObjectChangesToCommit {
+                            changed_objects: move_objects.clone(),
+                            deleted_objects: vec![],
+                        };
+                        info!(
+                            "Start persisting {} objects to objects table from {}",
+                            live_obj_cnt, object_file_path
+                        );
+                        store_clone
+                            .persist_objects(vec![object_changes])
+                            .await
+                            .expect("Failed to persist to objects from restore");
+                        info!(
+                            "Finished persisting {} objects to objects table from {}",
+                            live_obj_cnt, object_file_path
+                        );
+
+                        let objects_snapshot_changes = TransactionObjectChangesToCommit {
+                            changed_objects: move_objects,
+                            deleted_objects: vec![],
+                        };
+                        store_clone
+                            .persist_objects_snapshot(vec![objects_snapshot_changes])
+                            .await
+                            .expect("Failed to persist objects snapshot");
+
+                        bar_clone.inc(1);
+                        bar_clone.set_message(format!(
+                            "Restored {} live move objects from {}",
+                            live_obj_cnt, object_file_path
+                        ));
+                        Ok::<(), anyhow::Error>(())
+                    });
+                    restore_tasks.push(restore_task);
+                }
+
+                for restore_task in restore_tasks {
+                    restore_task.await??;
+                }
```

Comment:
I believe here you can use `futures::future::join_all(restore_tasks).await`.. unless you're separating them to make error handling easier?